modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -138,11 +275,9 @@ class CV_solvePnPRansac_Test : public cvtest::BaseTest
         bool isTestSuccess = inliers.size() >= points.size()*0.95;
 
         double rvecDiff = cvtest::norm(rvec, trueRvec, NORM_L2), tvecDiff = cvtest::norm(tvec, trueTvec, NORM_L2);
-        isTestSuccess = isTestSuccess && rvecDiff < epsilon[method] && tvecDiff < epsilon[method];
-        double error = rvecDiff > tvecDiff ? rvecDiff : tvecDiff;
-        //cout << error << " " << inliers.size() << " " << eps[method] << endl;
-        if (error > maxError)
-            maxError = error;
+        isTestSuccess = isTestSuccess && rvecDiff < eps[method] && tvecDiff < eps[method];
```

Comment:
Ultimately, I think that we should have different thresholds for translation and rotation. Pratically, we use the same values...

Reply:
I believe, robust test check should call `projectPoints()` with estimated rvec/tvec and check relative norm L2.